CMakeLists.txt

```diff
@@ -63,10 +63,18 @@ string(REGEX REPLACE  "-std=[^ ]+" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 if(DEFINED ENV{AR})
     set(CMAKE_AR $ENV{AR})
 endif()
-# Make CMAKE read CPPFLAGS
-set(CMAKE_CXX_FLAGS "$ENV{CPPFLAGS} ${CMAKE_CXX_FLAGS}")
+
 # Add default project CXX flags
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -fPIC -O3 -Wall -ftree-vectorize -g")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -fPIC -Wall -ftree-vectorize")
```

Comment:
@leezu @tgaddair FYI, I think we used to always build with debug symbols "-g" before, even in release mode, which gives extra stacktrace info when program crashes. Downside of "-g" is bigger libraries.

Reply:
Though `CMAKE_BUILD_TYPE=Release` generally does not include debug information. If the previous behaviour is desired, we can consider defaulting to the  `CMAKE_BUILD_TYPE=RelWithDebInfo` build type inside `setup.py`. WDYT? https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html

Reply:
I think having the stack trace is a very useful feature, so `CMAKE_BUILD_TYPE=RelWithDebInfo` sounds good to me.  What do you think @nvcastet ?

Reply:
yes I agree with RelWithDebInfo

Reply:
Awesome, @leezu can you put together a PR?

Reply:
@nvcastet I'm actually not sure how the `self.debug` attribute is set in setup.py:
https://github.com/horovod/horovod/blob/00dfae090777ba27d4ea768d7a92ba3760a247f5/setup.py#L64

Can you clarify? Essentially we just need to change the default value of this line to `RelWithDebInfo`

Reply:
yes default value to `RelWithDebInfo`. To be honest i don't know how `self.debug` get set to `True`. I kept the code that was there before my PR.